workflows/simple_workflow.go

```diff
@@ -20,4 +20,13 @@ func (simpleWorkflow *SimpleWorkflow) RunVirt() {
 		node.ProcessVirt(predecessorNodeResults)
 		predecessorNodeResults = append(predecessorNodeResults, node.GetResult()) 
 	}
-}
\ No newline at end of file
+}
+
+func (simpleWorkflow *SimpleWorkflow) GetResult() map[string]interface{} {
+	var nodeToResult map[string]interface{} = make(map[string]interface{})
+	for _, node := range simpleWorkflow.Nodes {
+		nodeToResult[node.GetName()] = node.GetResult()
```

Comment:
This is dangerous, because it's possible to have the same node appear multiple times in the same workflow. They'd have the same node.name, right?

Reply:
The python version only stores node results for the purpose of giving node n the results of nodes 0 to n-1 so I think I do not need this function. Removed it,

Reply:
I think getting the results of all nodes in a workflow is needed for testing (as in the Python version the only function that calls a workflow's getResult is the getPredecessorResults function).  I added a GetNodes function in the workflow just to make testing to see node's results possible (it would be nice if there was an @visiblefortesting in golang).

Reply:
I added a GetNodes function in the workflow to make seeing node's results possible for testing purposes only (it would be nice if there was a JAVA @VisibleForTesting in golang).

Reply:
The  GetNodes function in the Workflow class can also be used to get the nodes results as we do in main.go.